shuup/core/models/_media.py

```diff
@@ -30,7 +31,24 @@ class MediaFolder(models.Model):
     folder = FilerFolderField(related_name="media_folder", verbose_name=_("folder"), on_delete=models.CASCADE)
     shops = models.ManyToManyField(
         "shuup.Shop", related_name="media_folders", verbose_name=_('shops'),
-        help_text=_("Select which shops you would like the folder to be visible in."))
+        help_text=_("Select which shops you would like the folder to be visible in.")
+    )
+
+    user_access = models.ManyToManyField(
```

Comment:
So what happens if vendor/supplier are linked to multiple users?

Reply:
Then all the vendors users will have to be added here.